snowflake/pkg/snowflake/snowflake_test.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2022 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snowflake
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/go-logr/logr/testr"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const warehouseName = "ANTREA-QUERIES"
+
+func TestCreateWarehouse(t *testing.T) {
+	db, mock, err := sqlmock.New(sqlmock.QueryMatcherOption(sqlmock.QueryMatcherEqual))
+	require.NoErrorf(t, err, "an error was not expected when opening a stub database connection")
+	defer db.Close()
+
+	logger := testr.New(t)
+	c := NewClient(db, logger)
+	warehouseSize := WarehouseSizeType("XSMALL")
+	minClusterCount := int32(1)
+	maxClusterCount := int32(3)
+	scalingPolicy := ScalingPolicyStandard
+	autoSuspend := int32(60)
+	intiallySuspended := true
+	config := WarehouseConfig{
+		Size:               &warehouseSize,
+		MinClusterCount:    &minClusterCount,
+		MaxClusterCount:    &maxClusterCount,
+		ScalingPolicy:      &scalingPolicy,
+		AutoSuspend:        &autoSuspend,
+		InitiallySuspended: &intiallySuspended,
+	}
+	query := fmt.Sprintf("CREATE WAREHOUSE %s WITH WAREHOUSE_SIZE = XSMALL MIN_CLUSTER_COUNT = 1 MAX_CLUSTER_COUNT = 3 SCALING_POLICY = STANDARD AUTO_SUSPEND = 60 INITIALLY_SUSPENDED = true", warehouseName)
```

Comment:
Will it be better to format these variables `warehouseSize`, ..., `intiallySuspended` as well in this string?

Reply:
yes! thanks for catching it